### PR TITLE
Feature/178928604 enable access for private experiments

### PR DESCRIPTION
**Pull request by upendrakumbham**
This feature allows users to see private experiments if they know the access key in the single cell.
Companion PR: [atlas-web-core](https://github.com/ebi-gene-expression-group/atlas-web-core/pull/66)

### CODE REVIEW

File: app/src/test/java/uk/ac/ebi/atlas/experimentpage/cellplot/JsonCellPlotControllerWIT.java [hunk 1]
```diff
@@ -90,10 +84,8 @@ void cleanDatabaseTables() {
     @BeforeEach
     void setUp() {
         var experimentAccession = jdbcUtils.fetchRandomPublicExperimentAccession();
-        var experiment = experimentTrader.getPublicExperiment(experimentAccession);
-
-        experimentCrud.updateExperimentPrivate(experimentAccession, true);
-        accessKey = experiment.getAccessKey();
+        jdbcUtils.updatePublicExperimentAccessionToPrivate(experimentAccession);
+        accessKey = jdbcUtils.fetchExperimentAccessKey(experimentAccession);
```
Comment (upendrakumbham) [inline app/src/test/java/uk/ac/ebi/atlas/experimentpage/cellplot/JsonCellPlotControllerWIT.java:89]: Hi @alfonsomunozpomer - we are thinking that these lines are                                                                        
```
jdbcUtils.updatePublicExperimentAccessionToPrivate(experimentAccession);
 accessKey = jdbcUtils.fetchExperimentAccessKey(experimentAccession); 
```  
 better than the previous lines as we are not calling the actual codebase to get test data inputs.

I couldn't find any suitable method to replace this` expetedCluster() `with JDBCutils. I could see `fetchKsFromCellGroups`, but expected outcome is a `preferredK`.

Please suggest us if you have other opinion on this, we can discuss.

File: app/src/main/java/uk/ac/ebi/atlas/configuration/CacheConfig.java [hunk 2]
```diff
@@ -42,6 +42,7 @@ public CacheManager cacheManager() {
 
                 builder -> builder.name("hcaMetadata"),
                 builder -> builder.name("inferredCellTypesOntology"),
-                builder -> builder.name("inferredCellTypesAuthors"));
+                builder -> builder.name("inferredCellTypesAuthors"),
+                builder -> builder.name("experimentAccessions"));
```
Comment (haideriqbal) [inline app/src/main/java/uk/ac/ebi/atlas/configuration/CacheConfig.java:46]: I guess a better name for this cache would be `privateExperimentAccessions`

Comment (upendrakumbham) [reply app/src/main/java/uk/ac/ebi/atlas/configuration/CacheConfig.java:46]: yeah, Thanks. I will fix it.

Comment (upendrakumbham) [reply app/src/main/java/uk/ac/ebi/atlas/configuration/CacheConfig.java]: Fixed it.

**Review by haideriqbal — CHANGES_REQUESTED** [review]
We will need to evict the new `privateExperimentAcession` cache when performing CRUD operations. The change needs to in this file https://github.com/ebi-gene-expression-group/atlas-web-single-cell/blob/develop/app/src/main/java/uk/ac/ebi/atlas/experimentimport/ScxaExperimentCrud.java

**Comment by upendrakumbham** [comment]
> We will need to evict the new `privateExperimentAcession` cache when performing CRUD operations. The change needs to in this file https://github.com/ebi-gene-expression-group/atlas-web-single-cell/blob/develop/app/src/main/java/uk/ac/ebi/atlas/experimentimport/ScxaExperimentCrud.java

Ok, I will fix this one.

File: app/src/main/java/uk/ac/ebi/atlas/experimentimport/admin/SingleCellExperimentAdminController.java [hunk 3]
```diff
@@ -17,4 +20,11 @@ public SingleCellExperimentAdminController(DataFileHub dataFileHub,
                         new ExperimentOpLogWriter(dataFileHub),
                         new SingleCellOpsExecutionService(scxaExperimentCrud)));
     }
+
+    @Caching(evict = {
+            @CacheEvict(cacheNames = "privateExperimentAccessions", allEntries = true)})
+    @GetMapping(value = "/json/experiments/private/experimentAccessions/clearCache")
```
Comment (haideriqbal) [inline app/src/main/java/uk/ac/ebi/atlas/experimentimport/admin/SingleCellExperimentAdminController.java:26]: It's a minor change but i believe we can simplify the endpoint here. As it is just to evict private experiment cache and not returning anything we can rename it as `/private/experiments/cache/clear`

File: app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java [hunk 4]
```diff
@@ -34,7 +35,7 @@
 @WebAppConfiguration
 @ContextConfiguration(classes = TestConfig.class)
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
-class ExperimentPageContentServiceIT {
+    class ExperimentPageContentServiceIT {
```
Comment (haideriqbal) [inline app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java:38]: unnecessary ident here 👀

**Review by haideriqbal — DISMISSED** [review]
A couple of minor changes other than that LGTM!

**Comment by upendrakumbham** [comment]
We have this PR https://github.com/ebi-gene-expression-group/atlas-web-core/pull/70 to follow after rebase.

**Pull request closed without merging**